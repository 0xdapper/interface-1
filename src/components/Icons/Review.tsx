import { Edit } from 'react-feather'
import styled from 'styled-components'

const ReviewIconContainer = styled.div`
  background-color: #fc72ff;
  border-radius: 100px;
  width: 24px;
  height: 24px;
  text-align: center;
`
export const Review = () => {
  return (
    <ReviewIconContainer>
      <Edit size="12px" strokeWidth="2px" />
    </ReviewIconContainer>
  )
}
