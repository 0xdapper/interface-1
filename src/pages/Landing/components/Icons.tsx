export function PriceArrowUp() {
  return (
    <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        d="M10.868 7.17012L13.7881 11.4985C14.2788 12.226 13.8628 13.3334 13.0988 13.3334H6.90123C6.1372 13.3334 5.72122 12.2252 6.21191 11.4985L9.13196 7.17012C9.58465 6.49928 10.4154 6.49928 10.868 7.17012Z"
        fill="#40B66B"
      />
    </svg>
  )
}

export function PriceArrowDown() {
  return (
    <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        d="M10.868 12.8304L13.7881 8.50196C14.2788 7.77446 13.8628 6.66704 13.0988 6.66704H6.90123C6.1372 6.66704 5.72122 7.7753 6.21191 8.50196L9.13196 12.8304C9.58465 13.5012 10.4154 13.5012 10.868 12.8304Z"
        fill="#FD766B"
      />
    </svg>
  )
}
