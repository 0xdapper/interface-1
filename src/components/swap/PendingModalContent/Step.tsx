import { CheckMark } from 'components/Icons/CheckMark'
import { LoaderV3 } from 'components/Icons/LoadingSpinner'
import Row, { RowBetween } from 'components/Row'
import { ReactElement, useEffect, useState } from 'react'
import styled, { Keyframes, keyframes } from 'styled-components'
import { ThemedText } from 'theme/components'

export interface StepDetails {
  icon: ReactElement
  rippleColor?: string
  previewTitle: string
  actionRequiredTitle: string | ReactElement
  inProgressTitle?: string
  delayedTitle?: string
  delayedStartTitle?: string
  delayedEndTitle?: string
  timerValueInSeconds?: number
  timeToStart?: number
  timeToEnd?: number
  learnMoreLinkText?: string
  learnMoreLinkHref?: string
}

export enum StepStatus {
  PREVIEW,
  ACTIVE,
  IN_PROGRESS,
  COMPLETE,
}

const outerRingAnimation = keyframes`
  0% {
    transform: scale(1.4);
    opacity: 0.25;
  }
  50% {
    transform: scale(1.4);
    opacity: 0.75;
  }
  100% {
    transform: scale(1);
    opacity: 0;
  }
`
const innerRingAnimation = keyframes`
  0% {
    transform: scale(1);
    opacity: 0.25;
  }
  30% {
    transform: scale(1.3);
    opacity: 1;
  }
  60% {
    transform: scale(1.3);
    opacity: 0.4;
  }
  100% {
    transform: scale(1);
    opacity: 0.3;
  }
`
const Ring = styled.div<{ width: string; color: string; animation: Keyframes }>`
  position: absolute;
  width: 24px;
  height: 24px;
  border: ${({ width }) => width} solid ${({ color }) => color};
  border-radius: 50%;
  animation: ${({ animation }) => animation} 2s linear infinite;
`
const IconWrapper = styled.div<{ isActive: boolean }>`
  width: 24px;
  height: 24px;
  filter: ${({ isActive }) => `grayscale(${isActive ? 0 : 1})`};
  opacity: ${({ isActive }) => (isActive ? '1' : '0.5')};
`
function RippleAnimation({ isActive, rippleColor }: { isActive: boolean; rippleColor?: string }) {
  if (!isActive || !rippleColor) {
    return null
  }
  return (
    <>
      <Ring width="1px" color={rippleColor} animation={innerRingAnimation} />
      <Ring width="0.5px" color={rippleColor} animation={outerRingAnimation} />
    </>
  )
}
function Icon({ stepStatus, icon, rippleColor }: { stepStatus: StepStatus; icon: ReactElement; rippleColor?: string }) {
  if (stepStatus === StepStatus.IN_PROGRESS) {
    return <LoaderV3 size="24px" fill={rippleColor} />
  }
  return (
    <>
      <RippleAnimation isActive={stepStatus === StepStatus.ACTIVE} rippleColor={rippleColor} />
      <IconWrapper isActive={stepStatus === StepStatus.ACTIVE}>{icon}</IconWrapper>
    </>
  )
}

function Title({
  stepStatus,
  stepDetails,
  isTimeRemaining,
}: {
  stepStatus: StepStatus
  stepDetails: StepDetails
  isTimeRemaining: boolean
}) {
  switch (stepStatus) {
    case StepStatus.PREVIEW:
      return <ThemedText.BodySecondary>{stepDetails.previewTitle}</ThemedText.BodySecondary>
    case StepStatus.ACTIVE:
      return isTimeRemaining ? (
        <ThemedText.BodyPrimary>{stepDetails.actionRequiredTitle}</ThemedText.BodyPrimary>
      ) : (
        <ThemedText.BodyPrimary>{stepDetails.delayedStartTitle}</ThemedText.BodyPrimary>
      )
    case StepStatus.IN_PROGRESS:
      return isTimeRemaining ? (
        <ThemedText.BodyPrimary>{stepDetails.inProgressTitle}</ThemedText.BodyPrimary>
      ) : (
        <ThemedText.BodyPrimary>{stepDetails.delayedEndTitle}</ThemedText.BodyPrimary>
      )
    case StepStatus.COMPLETE:
      return <ThemedText.BodySecondary>{stepDetails.previewTitle}</ThemedText.BodySecondary>
    default:
      return null
  }
}

function Timer({ secondsRemaining }: { secondsRemaining: number }) {
  const minutes = Math.floor(secondsRemaining / 60)
  const seconds = secondsRemaining % 60
  const minutesText = minutes < 10 ? `0${minutes}` : minutes
  const secondsText = seconds < 10 ? `0${seconds}` : seconds
  const timerText = `${minutesText}:${secondsText}`
  return <ThemedText.BodySecondary paddingRight="8px">{timerText}</ThemedText.BodySecondary>
}

export function Step({ stepStatus, stepDetails }: { stepStatus: StepStatus; stepDetails: StepDetails }) {
  // Timer is shown in three cases:
  // (1) User has a specified amount of time to perform a required action. Timer starts running as soon as the step becomes active.
  // (2) Step has an estimated amount of time in which it should be completed. Timer is set but not running when step becomes active.
  // (3) Step has an estimated amount of time in which it should be completed. Timer starts running when step is in progress.
  const isTimed =
    (stepStatus === StepStatus.ACTIVE && Boolean(stepDetails.timeToStart)) ||
    (stepStatus === StepStatus.ACTIVE && Boolean(stepDetails.timeToEnd)) ||
    (stepStatus === StepStatus.IN_PROGRESS && Boolean(stepDetails.timeToEnd))
  const [secondsRemaining, setSecondsRemaining] = useState(0)

  useEffect(() => {
    if (stepStatus === StepStatus.ACTIVE && Boolean(stepDetails.timeToStart)) {
      setSecondsRemaining(stepDetails.timeToStart ?? 0)
    } else if (stepStatus === StepStatus.ACTIVE && Boolean(stepDetails.timeToEnd)) {
      setSecondsRemaining(stepDetails.timeToEnd ?? 0)
      return
    } else if (stepStatus === StepStatus.IN_PROGRESS) {
      setSecondsRemaining(stepDetails.timeToEnd ?? 0)
    } else {
      return
    }

    const timer = setInterval(() => {
      setSecondsRemaining((prevSecondsRemaining) => {
        if (prevSecondsRemaining > 0) {
          return prevSecondsRemaining - 1
        }
        clearInterval(timer)
        return 0
      })
    }, 1000)

    return () => clearInterval(timer)
  }, [stepStatus, stepDetails.timeToStart, stepDetails.timeToEnd])

  return (
    <RowBetween>
      <Row align="center" gap="12px">
        <Icon stepStatus={stepStatus} icon={stepDetails.icon} rippleColor={stepDetails.rippleColor} />
        <Title stepStatus={stepStatus} stepDetails={stepDetails} isTimeRemaining={isTimed && secondsRemaining > 0} />
      </Row>
      {isTimed && <Timer secondsRemaining={secondsRemaining} />}
      {stepStatus === StepStatus.COMPLETE && <CheckMark />}
    </RowBetween>
  )
}
