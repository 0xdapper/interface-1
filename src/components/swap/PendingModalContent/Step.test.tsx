import { Swap } from 'components/Icons/Swap'
import { act, render, screen } from 'test-utils/render'

import { Step, StepStatus } from './Step'

describe('Step in swap confirmation flow', () => {
  const stepDetails = {
    icon: <Swap />,
    rippleColor: '#4C82FB',
    previewTitle: 'Confirm swap',
    actionRequiredTitle: 'Confirm swap in wallet',
    inProgressTitle: 'Swap pending...',
    learnMoreLinkText: 'What is a transaction?',
    learnMoreLinkHref: 'https://support.uniswap.org/what-is-a-transaction',
  }

  it('displays an upcoming step', () => {
    const { asFragment } = render(<Step stepStatus={StepStatus.PREVIEW} stepDetails={stepDetails} />)

    expect(asFragment()).toMatchSnapshot()
    // Icon is shown and grayed out
    expect(screen.getByTestId('step-icon')).toHaveStyleRule('filter', 'grayscale(1)')
    // No ripple animation
    expect(screen.queryByTestId('icon-ripple-animation')).not.toBeInTheDocument()
    // Preview title shown
    expect(screen.getByText(stepDetails.previewTitle)).toBeInTheDocument()
    // No timer or checkmark icon
    expect(screen.queryByTestId('step-timer')).not.toBeInTheDocument()
    expect(screen.queryByTestId('checkmark-icon')).not.toBeInTheDocument()
  })

  it('displays an active step, awaiting user action - not timed and no ETA', () => {
    const { asFragment } = render(<Step stepStatus={StepStatus.ACTIVE} stepDetails={stepDetails} />)

    expect(asFragment()).toMatchSnapshot()
    // Icon is shown and not grayed out
    expect(screen.getByTestId('step-icon')).toHaveStyleRule('filter', 'grayscale(0)')
    // Ripple animation is active
    expect(screen.getByTestId('icon-ripple-animation').firstChild).toHaveAttribute('color', stepDetails.rippleColor)
    // Action Required title is shown
    expect(screen.getByText(stepDetails.actionRequiredTitle)).toBeInTheDocument()
    // No timer or checkmark icon
    expect(screen.queryByTestId('step-timer')).not.toBeInTheDocument()
    expect(screen.queryByTestId('checkmark-icfon')).not.toBeInTheDocument()
  })

  it('displays an active step, awaiting user action - timed', async () => {
    const TIME_TO_START = 20
    const DELAYED_START_TITLE = 'Confirmation timed out. Please retry.'
    jest.useFakeTimers()

    const { asFragment } = render(
      <Step
        stepStatus={StepStatus.ACTIVE}
        stepDetails={{
          ...stepDetails,
          timeToStart: TIME_TO_START,
          delayedStartTitle: DELAYED_START_TITLE,
        }}
      />
    )

    expect(asFragment()).toMatchSnapshot()
    // Icon is shown and not grayed out
    expect(screen.getByTestId('step-icon')).toHaveStyleRule('filter', 'grayscale(0)')
    // Ripple animation is active
    expect(screen.getByTestId('icon-ripple-animation').firstChild).toHaveAttribute('color', stepDetails.rippleColor)
    // Action Required title is shown
    expect(screen.getByText(stepDetails.actionRequiredTitle)).toBeInTheDocument()
    // Timer is shown and functional
    expect(screen.getByText('00:20')).toBeInTheDocument()
    act(() => {
      jest.advanceTimersByTime(5000)
    })
    expect(screen.getByText('00:15')).toBeInTheDocument()
    act(() => {
      jest.advanceTimersByTime(15000)
    })
    expect(screen.getByTestId('step-timer')).toHaveTextContent('00:00')
    expect(screen.getByText(DELAYED_START_TITLE)).toBeInTheDocument()
    expect(screen.queryByTestId('checkmark-icon')).not.toBeInTheDocument()
  })

  it('displays an active step, ETA only', async () => {
    const TIME_TO_END = 20
    const DELAYED_END_TITLE = 'Longer than expected...'
    jest.useFakeTimers()

    const { asFragment } = render(
      <Step
        stepStatus={StepStatus.ACTIVE}
        stepDetails={{
          ...stepDetails,
          timeToEnd: TIME_TO_END,
          delayedEndTitle: DELAYED_END_TITLE,
        }}
      />
    )

    expect(asFragment()).toMatchSnapshot()
    // Icon is shown and not grayed out
    expect(screen.getByTestId('step-icon')).toHaveStyleRule('filter', 'grayscale(0)')
    // Ripple animation is active
    expect(screen.getByTestId('icon-ripple-animation').firstChild).toHaveAttribute('color', stepDetails.rippleColor)
    // Action Required title is shown
    expect(screen.getByText(stepDetails.actionRequiredTitle)).toBeInTheDocument()
    // Timer is shown and preview only
    expect(screen.getByText('00:20')).toBeInTheDocument()
    act(() => {
      jest.advanceTimersByTime(5000)
    })
    expect(screen.getByText('00:20')).toBeInTheDocument()
    expect(screen.queryByText(DELAYED_END_TITLE)).not.toBeInTheDocument()
    expect(screen.queryByTestId('checkmark-icon')).not.toBeInTheDocument()
  })

  it('displays an active step in progress', async () => {
    const TIME_TO_END = 20
    const DELAYED_END_TITLE = 'Longer than expected...'
    jest.useFakeTimers()

    const { asFragment } = render(
      <Step
        stepStatus={StepStatus.IN_PROGRESS}
        stepDetails={{
          ...stepDetails,
          timeToEnd: TIME_TO_END,
          delayedEndTitle: DELAYED_END_TITLE,
        }}
      />
    )

    expect(asFragment()).toMatchSnapshot()
    // Loader icon is shown
    expect(screen.getByTestId('loader-icon')).toHaveAttribute('fill', stepDetails.rippleColor)
    // Action Required title is shown
    expect(screen.getByText(stepDetails.inProgressTitle)).toBeInTheDocument()
    // Timer is shown and functional
    expect(screen.getByText('00:20')).toBeInTheDocument()
    act(() => {
      jest.advanceTimersByTime(5000)
    })
    expect(screen.getByText('00:15')).toBeInTheDocument()
    act(() => {
      jest.advanceTimersByTime(15000)
    })
    expect(screen.getByTestId('step-timer')).toHaveTextContent('00:00')
    expect(screen.getByText(DELAYED_END_TITLE)).toBeInTheDocument()
    expect(screen.queryByTestId('checkmark-icon')).not.toBeInTheDocument()
  })

  it('displays checkmark once step is complete', () => {
    const { asFragment } = render(<Step stepStatus={StepStatus.COMPLETE} stepDetails={stepDetails} />)

    expect(asFragment()).toMatchSnapshot()
    expect(screen.getByTestId('step-icon')).toBeInTheDocument()
    expect(screen.getByText(stepDetails.previewTitle)).toBeInTheDocument()
    expect(screen.queryByTestId('step-timer')).not.toBeInTheDocument()
    expect(screen.getByTestId('checkmark-icon')).toBeInTheDocument()
  })
})
