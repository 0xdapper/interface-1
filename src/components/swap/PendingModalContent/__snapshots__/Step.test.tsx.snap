// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step in swap confirmation flow displays an active step in progress 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 12px;
}

.c2 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  color: #222222;
}

.c7 {
  color: #7D7D7D;
}

.c4 {
  height: 24px;
  width: 24px;
}

.c4 path {
  stroke: #CECECE;
  background: #7D7D7D;
  fill: #4C82FB;
}

.c5 {
  -webkit-animation: 2s fvtopB linear infinite;
  animation: 2s fvtopB linear infinite;
}

<div
    class="c0 c1 c2"
  >
    <div
      class="c0 c3"
    >
      <svg
        class="c4 c5"
        data-testid="loader-icon"
        fill="#4C82FB"
        stroke="#CECECE"
        viewBox="0 0 54 54"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M53.6666 26.9999C53.6666 41.7275 41.7276 53.6666 27 53.6666C12.2724 53.6666 0.333313 41.7275 0.333313 26.9999C0.333313 12.2723 12.2724 0.333252 27 0.333252C41.7276 0.333252 53.6666 12.2723 53.6666 26.9999ZM8.33331 26.9999C8.33331 37.3092 16.6907 45.6666 27 45.6666C37.3093 45.6666 45.6666 37.3092 45.6666 26.9999C45.6666 16.6906 37.3093 8.33325 27 8.33325C16.6907 8.33325 8.33331 16.6906 8.33331 26.9999Z"
          fill="#CECECE"
          opacity="0.1"
        />
        <path
          d="M49.6666 26.9999C51.8758 26.9999 53.6973 25.1992 53.3672 23.0149C53.0452 20.884 52.4652 18.7951 51.6368 16.795C50.2966 13.5597 48.3324 10.62 45.8562 8.14374C43.3799 5.66751 40.4402 3.70326 37.2049 2.36313C35.2048 1.53466 33.1159 0.954747 30.985 0.632693C28.8007 0.30256 27 2.12411 27 4.33325C27 6.54239 28.8108 8.29042 30.9695 8.76019C32.0523 8.99585 33.1146 9.32804 34.1434 9.75417C36.4081 10.6923 38.4659 12.0672 40.1993 13.8006C41.9327 15.534 43.3076 17.5918 44.2457 19.8565C44.6719 20.8853 45.004 21.9476 45.2397 23.0304C45.7095 25.1891 47.4575 26.9999 49.6666 26.9999Z"
          fill="#CECECE"
        />
      </svg>
      <div
        class="c6 css-1urox24"
      >
        Swap pending...
      </div>
    </div>
    <div
      class="c7 css-15d6nrs"
      data-testid="step-timer"
    >
      00:20
    </div>
  </div>
</DocumentFragment>
`;

exports[`Step in swap confirmation flow displays an active step, ETA only 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 12px;
}

.c2 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 {
  color: #222222;
}

.c8 {
  color: #7D7D7D;
}

.c4 {
  position: absolute;
  width: 24px;
  height: 24px;
  border: 1px solid #4C82FB;
  border-radius: 50%;
  -webkit-animation: cdcKrq 2s linear infinite;
  animation: cdcKrq 2s linear infinite;
}

.c5 {
  position: absolute;
  width: 24px;
  height: 24px;
  border: 0.5px solid #4C82FB;
  border-radius: 50%;
  -webkit-animation: fUGyYy 2s linear infinite;
  animation: fUGyYy 2s linear infinite;
}

.c6 {
  width: 24px;
  height: 24px;
  -webkit-filter: grayscale(0);
  filter: grayscale(0);
  opacity: 1;
}

<div
    class="c0 c1 c2"
  >
    <div
      class="c0 c3"
    >
      <div>
        <div
          data-testid="icon-ripple-animation"
        >
          <div
            class="c4"
            color="#4C82FB"
            width="1px"
          />
          <div
            class="c5"
            color="#4C82FB"
            width="0.5px"
          />
        </div>
        <div
          class="c6"
          data-testid="step-icon"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="#4C82FB"
              height="24"
              rx="12"
              width="24"
            />
            <path
              d="M10.6663 8.33325L13.6663 5.33325M13.6663 5.33325L16.6663 8.33325M13.6663 5.33325V11.9999M13.333 15.6666L10.333 18.6666M10.333 18.6666L7.33301 15.6666M10.333 18.6666L10.333 12.6666"
              stroke="#F5F6FC"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.33333"
            />
          </svg>
        </div>
      </div>
      <div
        class="c7 css-1urox24"
      >
        Confirm swap in wallet
      </div>
    </div>
    <div
      class="c8 css-15d6nrs"
      data-testid="step-timer"
    >
      00:20
    </div>
  </div>
</DocumentFragment>
`;

exports[`Step in swap confirmation flow displays an active step, awaiting user action - not timed and no ETA 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 12px;
}

.c2 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 {
  color: #222222;
}

.c4 {
  position: absolute;
  width: 24px;
  height: 24px;
  border: 1px solid #4C82FB;
  border-radius: 50%;
  -webkit-animation: cdcKrq 2s linear infinite;
  animation: cdcKrq 2s linear infinite;
}

.c5 {
  position: absolute;
  width: 24px;
  height: 24px;
  border: 0.5px solid #4C82FB;
  border-radius: 50%;
  -webkit-animation: fUGyYy 2s linear infinite;
  animation: fUGyYy 2s linear infinite;
}

.c6 {
  width: 24px;
  height: 24px;
  -webkit-filter: grayscale(0);
  filter: grayscale(0);
  opacity: 1;
}

<div
    class="c0 c1 c2"
  >
    <div
      class="c0 c3"
    >
      <div>
        <div
          data-testid="icon-ripple-animation"
        >
          <div
            class="c4"
            color="#4C82FB"
            width="1px"
          />
          <div
            class="c5"
            color="#4C82FB"
            width="0.5px"
          />
        </div>
        <div
          class="c6"
          data-testid="step-icon"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="#4C82FB"
              height="24"
              rx="12"
              width="24"
            />
            <path
              d="M10.6663 8.33325L13.6663 5.33325M13.6663 5.33325L16.6663 8.33325M13.6663 5.33325V11.9999M13.333 15.6666L10.333 18.6666M10.333 18.6666L7.33301 15.6666M10.333 18.6666L10.333 12.6666"
              stroke="#F5F6FC"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.33333"
            />
          </svg>
        </div>
      </div>
      <div
        class="c7 css-1urox24"
      >
        Confirm swap in wallet
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Step in swap confirmation flow displays an active step, awaiting user action - timed 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 12px;
}

.c2 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 {
  color: #222222;
}

.c8 {
  color: #7D7D7D;
}

.c4 {
  position: absolute;
  width: 24px;
  height: 24px;
  border: 1px solid #4C82FB;
  border-radius: 50%;
  -webkit-animation: cdcKrq 2s linear infinite;
  animation: cdcKrq 2s linear infinite;
}

.c5 {
  position: absolute;
  width: 24px;
  height: 24px;
  border: 0.5px solid #4C82FB;
  border-radius: 50%;
  -webkit-animation: fUGyYy 2s linear infinite;
  animation: fUGyYy 2s linear infinite;
}

.c6 {
  width: 24px;
  height: 24px;
  -webkit-filter: grayscale(0);
  filter: grayscale(0);
  opacity: 1;
}

<div
    class="c0 c1 c2"
  >
    <div
      class="c0 c3"
    >
      <div>
        <div
          data-testid="icon-ripple-animation"
        >
          <div
            class="c4"
            color="#4C82FB"
            width="1px"
          />
          <div
            class="c5"
            color="#4C82FB"
            width="0.5px"
          />
        </div>
        <div
          class="c6"
          data-testid="step-icon"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="#4C82FB"
              height="24"
              rx="12"
              width="24"
            />
            <path
              d="M10.6663 8.33325L13.6663 5.33325M13.6663 5.33325L16.6663 8.33325M13.6663 5.33325V11.9999M13.333 15.6666L10.333 18.6666M10.333 18.6666L7.33301 15.6666M10.333 18.6666L10.333 12.6666"
              stroke="#F5F6FC"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.33333"
            />
          </svg>
        </div>
      </div>
      <div
        class="c7 css-1urox24"
      >
        Confirm swap in wallet
      </div>
    </div>
    <div
      class="c8 css-15d6nrs"
      data-testid="step-timer"
    >
      00:20
    </div>
  </div>
</DocumentFragment>
`;

exports[`Step in swap confirmation flow displays an upcoming step 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 12px;
}

.c2 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  color: #7D7D7D;
}

.c4 {
  width: 24px;
  height: 24px;
  -webkit-filter: grayscale(1);
  filter: grayscale(1);
  opacity: 0.5;
}

<div
    class="c0 c1 c2"
  >
    <div
      class="c0 c3"
    >
      <div>
        <div
          class="c4"
          data-testid="step-icon"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="#4C82FB"
              height="24"
              rx="12"
              width="24"
            />
            <path
              d="M10.6663 8.33325L13.6663 5.33325M13.6663 5.33325L16.6663 8.33325M13.6663 5.33325V11.9999M13.333 15.6666L10.333 18.6666M10.333 18.6666L7.33301 15.6666M10.333 18.6666L10.333 12.6666"
              stroke="#F5F6FC"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.33333"
            />
          </svg>
        </div>
      </div>
      <div
        class="c5 css-1urox24"
      >
        Confirm swap
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Step in swap confirmation flow displays checkmark once step is complete 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 12px;
}

.c2 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  color: #7D7D7D;
}

.c4 {
  width: 24px;
  height: 24px;
  -webkit-filter: grayscale(1);
  filter: grayscale(1);
  opacity: 0.5;
}

<div
    class="c0 c1 c2"
  >
    <div
      class="c0 c3"
    >
      <div>
        <div
          class="c4"
          data-testid="step-icon"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="#4C82FB"
              height="24"
              rx="12"
              width="24"
            />
            <path
              d="M10.6663 8.33325L13.6663 5.33325M13.6663 5.33325L16.6663 8.33325M13.6663 5.33325V11.9999M13.333 15.6666L10.333 18.6666M10.333 18.6666L7.33301 15.6666M10.333 18.6666L10.333 12.6666"
              stroke="#F5F6FC"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.33333"
            />
          </svg>
        </div>
      </div>
      <div
        class="c5 css-1urox24"
      >
        Confirm swap
      </div>
    </div>
    <svg
      data-testid="checkmark-icon"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.00002 17C8.99902 17 8.99802 17 8.99602 17C8.72902 16.999 8.47502 16.892 8.28802 16.702L4.28802 12.64C3.90002 12.246 3.90503 11.613 4.29903 11.226C4.69303 10.839 5.32502 10.843 5.71302 11.237L9.00602 14.581L18.294 5.29398C18.685 4.90298 19.317 4.90298 19.708 5.29398C20.099 5.68398 20.099 6.31798 19.708 6.70798L9.70801 16.708C9.52001 16.895 9.26502 17 9.00002 17Z"
        fill="#40B66B"
      />
    </svg>
  </div>
</DocumentFragment>
`;
